### PR TITLE
Share Kit for React Native

### PR DESCRIPTION
**Pull request by psdkumar**
Resolves https://github.com/hellobloom/share-kit/issues/51

### CODE REVIEW

**Comment by psdkumar** [comment]
This is how the rendered button looks.
<p align="center">
  
</p>

@JaceHensley This is as close as I could get to the button. Is this ok? 
The code related to this is [here](https://github.com/psdkumar/share-kit-reactnative/blob/master/src/RequestButton.tsx)

**Comment by psdkumar** [comment]
The svg is rendered correctly. The problem is with the overlap of `backgroundPattern` and `linearGradient`

**Comment by JaceHensley** [comment]
I ran `npm i` and `npm run storybook` but it's failing like this:

```
.../share-kit-reactnative/node_modules/enhanced-resolve/lib/Resolver.js:254
		const idxQuery = identifier.indexOf("?");
		                            ^

TypeError: identifier.indexOf is not a function
```

There's also `npm audit` warnings can you take care of those? (except the one coming a share-kit dep)

**Comment by JaceHensley** [comment]
As far as the styling goes:

- Button needs to be rounded
- The bg image should be started further to the right (the ripples start a little after the lock icon)
- The text+logo should be centered to the whole button not the just the space that's left over after the lock icon
  - Not sure why that's not happening based on the RN styles, but a different approach may be needed to get that centering correctly (`position: absolute` on the lock icon container or something)

image

**Comment by psdkumar** [comment]
@JaceHensley Follow the instuctions in [README](https://github.com/psdkumar/share-kit-reactnative/blob/master/demo/README.md) to run the storybook.

**Comment by psdkumar** [comment]
@JaceHensley Please tell me if you are able to run the storybook on your android device without any problems.

**Comment by pbteja1998** [comment]
@JaceHensley Can you please review this?

**Comment by psdkumar** [comment]
@JaceHensley Can you please review this?

**Comment by JaceHensley** [comment]
I'll be taking another look at this today, thanks for your patience!

**Comment by psdkumar** [comment]
Hey @JaceHensley. Did you get a chance to look at this?

**Comment by JaceHensley** [comment]
Not yet, I will be lookin at it today tho

File: src/RequestButton.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+const getLink = (data: RequestData, callbackUrl: string) =>
+    `https://bloom.co/download?request=${encode(JSON.stringify(data))}&callback-url=${encodeURIComponent(
```
Comment (JaceHensley) [inline src/RequestButton.tsx]: This needs to append the `?share-kit-from=button` query param to `data.url`. Making sure to also support when there is already other query params on the url. [Impl](https://github.com/hellobloom/share-kit/blob/master/src/renderRequestElement.ts#L27) in share-kit

**Comment by psdkumar** [comment]
Try [this](https://stackoverflow.com/a/48155800) once..

**Comment by psdkumar** [comment]
I tried changing the dependency to use local project earlier. It did not work for some reason. I will see if I can fix that. For now if you want to test, you can change that to `@pbteja-dev/share-kit-reactnative`

**Comment by JaceHensley** [comment]
Okay, that got me a little further along! But now it's saying my phone cannot connect to the development server. Seems like it's trying to point to `localhost:8081` on my phone, how can I configure it to point to the local IP of my laptop?

**Comment by psdkumar** [comment]
You have to configure your phone. Check the [docs](http://facebook.github.io/react-native/docs/running-on-device)

**Comment by psdkumar** [comment]
First JS Server(`npm run start`) needs to be run. Then connect the device as per the above link. Then `npm run android`

**Comment by pbteja1998** [comment]
@JaceHensley By the way, what timezone are you in?

**Comment by psdkumar** [comment]
@JaceHensley What timezone are you in?

**Comment by psdkumar** [comment]
b399210d-2bca-4d79-8795-3d49569eeb00

File: src/RequestButton.tsx (hunk 1 above)
Comment (psdkumar) [reply src/RequestButton.tsx]: > This needs to append the `?share-kit-from=button` query param to `data.url`. Making sure to also support when there is already other query params on the url. [Impl](https://github.com/hellobloom/share-kit/blob/master/src/renderRequestElement.ts#L27) in share-kit

Done

File: src/RequestButton.tsx [hunk 2]
```diff
@@ -18,6 +20,8 @@ const getLink = (data: RequestData, callbackUrl: string) =>
         callbackUrl
     )}`;
 
+const appendParams = (callbackUrl: string) => appendQuery(callbackUrl, 'share-kit-from=button');
```
Comment (JaceHensley) [inline src/RequestButton.tsx]: The query param actually goes on the `requestData.url` not `buttonCallbackUrl`, see [here](https://github.com/hellobloom/share-kit/blob/master/src/renderRequestElement.ts#L27)

**Comment by JaceHensley** [comment]
I'm trying to tweak things locally and see them in the storybook but I can't because the dependency is to a published version instead of the local version. And as you said it breaks when setting it to `../` in the demo project. And publishing a version just to test out a change isn't a viable dev cycle. 

Is there a reason the storybook has to be a separate package than the regular package? Like how share-kit/share-kit-react have it set up.

**Comment by JaceHensley** [comment]
So I think to fix the ripple effect you should swap the order of the `ImageBackground` and `LinearGradient` so that `ImageBackground` is the parent. And the `borderRadius` should probably go on the `TouchableOpacity`. Also we don't want that opacity on tap, not sure if there's a better component for this or if it should just set `activeOpacity` to 1.

**Comment by psdkumar** [comment]
> Is there a reason the storybook has to be a separate package than the regular package? Like how share-kit/share-kit-react have it set up. 

When I used `..` instead of the published package, the dependecies of `..` are not being installed automatically. So maybe if I add the dependencies manually, then may be we can use `..` instead of published version. Let me try it out once.

**Comment by psdkumar** [comment]
> I'm in PT, you?
IST

**Pull request closed without merging**